ci/travis/bazel.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
```

Comment:
Sure. It's for running `shellcheck` on `genrule`s. To do that it needs to call Bazel to output the `genrule` scripts after Bazel has processed them. Bazel then outputs them in its prototext format, which I parse here.

Reply:
I think it's probably easier to understand if you follow the logic starting in `main()` (which would be reading bottom-up). Let me know if I should add any explanations along the way.

Reply:
Oh, and it's probably worth mentioning: I imagine this file will probably get more Bazel-related functionality later on (in fact there are already things that I'm hoping to migrate from Bash to Bazel using this once it's merged), since invoking Bazel and parsing its output are common subfunctionality for lots of scripts. It's worth keeping that in mind when reviewing. It'll become more useful as time goes on.